src/MOIWrapper.jl

```diff
@@ -0,0 +1,558 @@
+        end
+        
+        addrow(row, lower, upper, instance, rows, cols, coefs)
+    end
+end
+
+"""
+    add_ranged_constraints!(m, A::CSRMatrix{Float64}, lb::Vector{Float64}, ub::Vector{Float64})::Void
+
+Adds linear constraints of the form `Ax (sense) rhs` to the model `m`.
+`sense` and `rhs` contain one element for each row in `A`.
+The `sense` is given by `backend_type(m, set)`.
+Ranged constraints (`set=MOI.Interval`) should be added via `add_ranged_constraint!` instead.
+See also: `LinQuadOptInterface.CSRMatrix`.
+"""
+function LQOI.add_ranged_constraints!(instance::Clp.ClpOptimizer, A::LinQuadOptInterface.CSRMatrix{Float64}, 
+        lb::Vector{Float64}, ub::Vector{Float64})
+
+    rows = A.row_pointers
+    cols = A.columns
+    coefs = A.coefficients
+           
+    nbrows = length(lb)
+    for row in 1:nbrows
+        lower = lb[row]
+        upper = ub[row]
+    
+        addrow(row, lower, upper, instance, rows, cols, coefs)                
+    end
+end
+
+
+"""
+get_rhs(m, row::Int)::Float64
+
+Get the right-hand side of the linear constraint in the 1-indexed row `row` in
+the model `m`.
+"""
+function LQOI.get_rhs(instance::ClpOptimizer, row::Int)::Float64
+    lower = replaceInf(get_row_lower(instance.inner))
+    upper = replaceInf(get_row_upper(instance.inner))
+    lb = lower[row]
+    ub = upper[row]
+    
+    if lb > -Inf
+        return lb
+    elseif ub < Inf
+        return ub
+    else
+        error("Either row_lower or row_upper must be of abs less than 1e20")
+    end
+end
+
+"""
+get_linear_constraint(m, row::Int)::Tuple{Vector{Int}, Vector{Float64}}
+
+Get the linear component of the constraint in the 1-indexed row `row` in
+the model `m`. Returns a tuple of `(cols, vals)`.
+"""
+function LQOI.get_linear_constraint(instance::ClpOptimizer, row::Int)::Tuple{Vector{Int}, Vector{Float64}}
+    A = get_constraint_matrix(instance.inner)
+    A_row = A[row,:]
+    return (Array{Int}(A_row.nzind), A_row.nzval)
+end
+
+"""
+change_objective_coefficient!(m, col, coef)
+
+Set the linear coefficient of the variable in column `col` to `coef` in the objective function.
+"""
+function LQOI.change_objective_coefficient!(instance::ClpOptimizer, col, coef)
+    objcoefs = get_obj_coefficients(instance.inner)
+    objcoefs[col] = coef
+    chg_obj_coefficients(instance.inner, objcoefs)
+end
+
+"""
+change_rhs_coefficient!(m, row, coef)
+
+Set the rhs of the constraint in row `row` to `coef`.
+"""
+function LQOI.change_rhs_coefficient!(instance::ClpOptimizer, row, coef)
+    lower = replaceInf(get_row_lower(instance.inner))
+    upper = replaceInf(get_row_upper(instance.inner))
+    lb = lower[row]
+    ub = upper[row]
+    
+    if lb > -Inf
+        lower[row] = coef
+        chg_row_lower(instance.inner, lower)
+    elseif ub < Inf
+        upper[row] = coef
+        chg_row_upper(instance.inner, upper)
+    else
+        error("Either row_lower or row_upper must be of abs less than 1e20")
```

Comment:
So when adding a constraint it allows bigger coefficients, but it truncates them to 1e20, but when modifying it doesn't? Confusing.